src/main/java/botimusprime/BotimusPrime.java

```diff
@@ -0,0 +1,74 @@
+package botimusprime;
+
+import botimusprime.parser.Parser;
+import botimusprime.storage.Storage;
+import botimusprime.tasks.TaskList;
+import botimusprime.ui.Ui;
+
+/**
+ * The BotimusPrime class is the main class for the BotimusPrime chatbot.
+ * It manages the main components of the chatbot, including storage, UI, task management,
+ * and command parsing. This class also contains the main method to start the chatbot.
+ */
+public class BotimusPrime {
+
+    private Storage storage;
+    private Ui ui;
+    private TaskList taskList;
+    private Parser parser;
+
+    /**
+     * Constructs a new BotimusPrime instance with the specified file name.
+     * Initializes the UI, storage, task list, and parser components.
+     *
+     * @param fileName The name of the file where tasks are stored.
+     */
+    public BotimusPrime(String fileName) {
+        this.ui = new Ui();
+        this.storage = new Storage(fileName);
+        taskList = storage.loadFromDisk();
+        parser = new Parser(taskList,ui, storage);
+    }
+
+    /**
+     * Returns the TaskList associated with this BotimusPrime instance.
+     *
+     * @return The TaskList containing all tasks managed by this instance.
+     */
+    public TaskList getTaskList() {
+        return taskList;
+    }
+
+    /**
+     * Starts the BotimusPrime chatbot.
+     * Displays a greeting message, then enters a loop to read and parse user commands.
+     * The loop continues until the user issues an exit command "bye".
+     */
+    public void run() {
```

Comment:
The method name run is a bit generic. While it is commonly used to start processes or applications, a more descriptive name like startChatbot could provide clearer insight into its purpose.

Reply:
Sounds good!